File: homework-05/src/main/java/ru/otus/homework/domain/Book.java
package ru.otus.homework.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

@Data
@AllArgsConstructor
@RequiredArgsConstructor
public class Book {
    private long id;

    @NonNull
    private String title;

    @NonNull
    private int year;

    @NonNull
    private Author author;

    @NonNull
    private Genre genre;
}

File: homework-11-ui/src/domain/Book.ts
import {Author} from "./Author";
import {Genre} from "./Genre";

export class Book {
    id: string;
    title: string;
    year: string;
    author: Author;
    genre: Genre;

    constructor(id: string, title: string, year: string, author: Author, genre: Genre) {
        this.id = id;
        this.title = title;
        this.year = year;
        this.author = author;
        this.genre = genre;
    }
}
